lang/cs/Org.Apache.REEF.Experimental/ParquetReader/ParquetReader.cs

```diff
@@ -0,0 +1,129 @@
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Experimental.ParquetReader.Parameters;
+
+namespace Org.Apache.REEF.Experimental.ParquetReader
+{
+    /// <summary>
+    /// Constructs a Parquet file reader.
+    /// </summary>
+    public class ParquetReader : IDisposable
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(ParquetReader));
+
+        private bool _disposed = false;
+
+        private class JavaProcessConfiguration
+        {
+            public const string javaExecutable = "java";
+            public const string mainClass = "org.apache.reef.experimental.parquet.ParquetReader";
+            public string ParquetPath { get; set; }
+            public string ClassPath { get; set; }
+        }
+
+        private class JavaProcess
+        {
+            public string AvroPath { get; set; }
+            public JavaProcessConfiguration Conf { get; set; }
+            public void Start()
+            {
+                var p = new Process();
+                p.StartInfo.FileName = JavaProcessConfiguration.javaExecutable;
+                p.StartInfo.Arguments =
+                    new[] { "-cp", Conf.ClassPath, JavaProcessConfiguration.mainClass, Conf.ParquetPath, AvroPath }
+                    .Aggregate((a, b) => a + ' ' + b);
+
+                Logger.Log(Level.Info, "Running Command: java {0}", p.StartInfo.Arguments);
+
+                p.Start();
+                p.WaitForExit();
+                p.Dispose();
+            }
+        }
+
+        private readonly JavaProcessConfiguration c;
+
+        /// <summary>
+        /// Constructor of ParquetReader for Tang Injection
+        /// </summary>
+        /// <param name="parquetPath">Path to input parquet file.</param>
+        /// <param name="jarPath">Path to jar file that contains Java parquet reader.</param>
+        [Inject]
+        private ParquetReader(
+            [Parameter(typeof(ParquetPathString))] string parquetPath,
+            [Parameter(typeof(ClassPathString))] string classPath)
+        {
+            if (!File.Exists(parquetPath))
+            {
+                throw new FileNotFoundException("Input parquet file {0} doesn't exist.", parquetPath);
+            }
+            c = new JavaProcessConfiguration
+            {
+                ParquetPath = parquetPath,
+                ClassPath = classPath
+            };
+        }
+
+        /// <summary>
+        /// Method to read the given parquet files.
+        /// </summary>
+        /// <returns>
+        /// Return a SequentialReader that can iterate data from each avro block.
+        /// </returns>
+        public SequentialReader<T> CreateSequentialReader<T>()
+        {
+            var p = new JavaProcess
+            {
+                AvroPath = Path.GetTempFileName(),
+                Conf = c
+            };
+
+            p.Start();
+
+            var stream = new FileStream(p.AvroPath, FileMode.Open);
```

Comment:
Why wouldn't the `Read` method return a class that is both `IEnumerable` and `IDisposable`? Then, its `Dispose` method can `.Dispose` the `stream`. Also, this would allow for multiple stream readers to be opened.